service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -97,13 +100,29 @@ public UUID createWorkspace(Workspace workspace) {
       jdbcTemplate.update(sql, params);
       logger.info("Inserted record for workspace {}", workspaceId);
     } catch (DuplicateKeyException e) {
-      throw new DuplicateWorkspaceException(
-          String.format(
-              "Workspace with id %s already exists - display name %s stage %s",
-              workspaceId,
-              workspace.getDisplayName().toString(),
-              workspace.getWorkspaceStage().toString()),
-          e);
+      // Workspace with workspace_id already exists.
+      if (e.getMessage()
+          .contains("duplicate key value violates unique constraint \"workspace_pkey\"")) {
```

Comment:
This looks brittle; a change to the DB vendor or version might break the message. Not sure what your alternatives are though.